File: python/src/leetcode/topinteviewquestions/ValidParenthesis.py
class Solution:
    def isValid(self, s: str) -> bool:
        tagmap = {")": "(", "}": "{", "]": "["}
        stack = []
        for char in s:
            if char in tagmap.keys():
                pop = ""
                if stack:
                    pop = stack.pop()

                if tagmap[char] != pop:
                    return False
            else:
                stack.append(char)
        return not stack

File: typescript/src/leetcode/2025/ValidParenthesis.ts
export function isValid(s: string): boolean {
  let stack: string[] = [];

  const brackets: { [key: string]: string } = {
    ")": "(",
    "}": "{",
    "]": "[",
  };

  for (let bracket of s) {
    // 閉じ括弧の場合はstackの一番上をチェック
    if (bracket in brackets) {
      if (stack.length === 0 || stack.pop() !== brackets[bracket]) {
        // 対応する括弧がなければfalse
        return false;
      }
    } else {
      // 一致しない場合は積む
      stack.push(bracket);
    }
  }
  return stack.length === 0;
}
